.github/workflows/auditbeat-macos-unit-tests.yml

```
@@ -0,0 +1,55 @@
+name: auditbeat-macos-unit-tests
+
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+    paths:
+      - auditbeat/
+      # OSS
+      - go.mod
+      - pytest.ini
+      - dev-tools/
+      - libbeat/**
+      - testing/**
+      # Workflow
+      - .github/workflows/auditbeat-macos-unit-tests.yml
+      - .github/workflows/actions/unit-test-composite-action/action.yml
+
+permissions:
+  contents: read
+
+## Concurrency only allowed in the main branch.
+## So old builds running for old commits within the same Pull Request are cancelled
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: ${{ github.ref != 'refs/heads/main' }}
+
+jobs:
+  unit-tests-push:
```

Maybe `branch` sounds a bit more accurate. 

```suggestion
  unit-tests-branch:
```